### ISSUE TITLE
`collapsible_match` generates invalid suggestion when `match` arm does not have `{}`

### ISSUE THREAD

**Issue by narpfel** [rust-lang/rust-clippy#16716]
### Summary

When `collapsible_match` is emitted for a `match` arm that is not a block, the suggestion removes the `i` from `if` and the closing `}`.

### Reproducer

Code:

```rust
pub fn f(n: u32) {
    match n {
        0 | 1 => if false { println!("hello world"); },
        _ => (),
    }
}
```

Current output:
```
warning: this `if` can be collapsed into the outer `match`
 --> <source>:3:18
  |
3 |         0 | 1 => if false { println!("hello world"); },
  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_match
  = note: `#[warn(clippy::collapsible_match)]` on by default
help: collapse nested if block
  |
3 -         0 | 1 => if false { println!("hello world"); },
3 +         0 | 1f false => { println!("hello world");,
  |

warning: 1 warning emitted
```

Desired output:
```
warning: this `if` can be collapsed into the outer `match`
 --> <source>:3:18
  |
3 |         0 | 1 => if false { println!("hello world"); },
  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_match
  = note: `#[warn(clippy::collapsible_match)]` on by default
help: collapse nested if block
  |
3 -         0 | 1 => if false { println!("hello world"); },
3 +         0 | 1 if false => { println!("hello world"); },
  |

warning: 1 warning emitted
```

Godbolt link: https://godbolt.org/z/9zeqs3rMb

### Version

```text
rustc 1.96.0-nightly (03749d625 2026-03-14)
binary: rustc
commit-hash: 03749d625404994801804ab9f2a539936c50bfea
commit-date: 2026-03-14
host: x86_64-unknown-linux-gnu
release: 1.96.0-nightly
LLVM version: 22.1.0
```
